File: web-clients/react/src/Components/MessageList/Message.js
import React from "react";
import PropTypes from "prop-types";
import VoteContainer from "./VoteContainer";
import { message as messageClass } from "./_style.css";

const Message = ({ id, created, text }) => {
  const dt = new Date(created).toLocaleString("en-US");

  return (
    <article className={messageClass}>
      <p>{text}</p>
      <footer>
        <VoteContainer subjectId={id} />
        <time dateTime={dt}>{dt}</time>
      </footer>
    </article>
  );
};

Message.propTypes = {
  id: PropTypes.number.isRequired,
  created: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

export default React.memo(Message);

File: server/messaging/MessageBoard.Messaging.Core/Message.cs
using System;

namespace MessageBoard.Messaging.Core
{
    public class Message
    {
        public long Id { get; set; }
        public DateTime Created { get; set; }
        public string Text { get; set; }

        public Message(long id, DateTime created, string text)
        {
            Id = id;
            Created = created;
            Text = text;
        }
    }
}
